article/app/services/dotcomponents/ArticlePicker.scala

```diff
@@ -78,14 +80,13 @@ object ArticlePicker {
     article100PercentPageFeatures.forall({ case (_, isMet) => isMet })
   }
 
-  def getTier(page: PageWithStoryPackage)(implicit request: RequestHeader): RenderType = {
+  def getTier(page: PageWithStoryPackage, path: String)(implicit
+      request: RequestHeader,
+  ): RenderType = {
     val checks = dcrChecks(page, request)
     val dcrCanRender = checks.values.forall(identity)
 
-    val tier =
-      if (request.forceDCROff) LocalRenderArticle
-      else if (request.forceDCR || dcrCanRender) RemoteRender
-      else LocalRenderArticle
+    val tier: RenderType = calculateTier(isPressed(ensureStartingForwardSlash(path)), dcrCanRender)
```

Comment:
`decideTier`?

'calculate' implies math to me

Reply:
Also, what's the value of this abstraction. Could we just keep this code inline?

Reply:
Yes, `decideTier` makes more sense indeed! We'll change that. As for the abstraction, it enabled us to easily write unit tests for this logic. Otherwise we would have to mock `PageWithStoryPackage`. This abstraction was suggested by Roberto as an easy / good way of handling the situation.

Reply:
So, and this is totally an aside, I have noticed that this approach tends to lead to a codebase that is heavily abstracted with lots of unit tests. Some people like this and I'm not saying it is wrong but, personally, I find this creates a situation where the code is hard to maintain. Changes or edits to code require tests to be updated and involve switching context between files and having to check to make sure you're not impacting other places in the code that might be importing the abstraction.

My own, personal preference is to favour integration tests where the interface that your testing is the same as the readers (or other consumers of the code like routes or apis requests) and then to keep code inline where it's closer to where it's being used and easier to maintain.

In general, Frontend seems like a unit test heavy codebase where abstractions are encouraged and so arguably it's right that this pattern be followed here. Maybe this type of code is better suited to it? My _point_ is, please don't take this as a rule and use it everywhere!